File: docs/src/pages/GuidePage/content/calculation/code/calculateProduct.ts
import { calculateProduct, CalculationOptions, LifeCycleModule, ImpactCategoryKey } from 'lcax'
import fs from 'fs'

const product_data = fs.readFileSync('product.json')
const product = JSON.parse(product_data)

const options: CalculationOptions = {
  referenceStudyPeriod: 50,
  lifeCycleModules: ['a1a3'] as LifeCycleModule[],
  impactCategories: ['gwp'] as ImpactCategoryKey[],
  overwriteExistingResults: true,
}

const impacts = calculateProduct(product, options)

File: docs/src/pages/GuidePage/content/calculation/code/calculate_product.py
from lcax import calculate_product, CalculationOptions, LifeCycleModule, ImpactCategoryKey, Product
import json

with open("product.json") as f:
    product = Product(**json.load(f))

options = CalculationOptions(
    reference_study_period=50,
    life_cycle_modules=[LifeCycleModule.A1A3],
    impact_categories=[ImpactCategoryKey.Gwp],
    overwrite_existing_results=True
)

impacts = calculate_product(product, options)
